### PR TITLE
Fix duplicate alias with liquid pattern

### LINKED ISSUE #4599: Alias duplication check failure
I created a content type which contained Title, Html Body and Alias. For the Alias I used the following pattern so that it will automatically generate the slug using the display_text.

`{{ ContentItem | display_text | slugify }}`

It works only when we have a different display_text for each content. If I create two different contents with same title, Orchard will allow me to create these contents and the system will generate the same alias for both of them. 

If we try to edit and save one of the contents, an error message (Your alias is already in use.) will appear on the screen informing the user about the duplication. This check failed during the content creation.

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes #4599 

Applies same logic as autoroute in generating unique aliases from liquid

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs [hunk 1]
```diff
@@ -53,20 +49,15 @@ public async override Task UpdatedAsync(UpdateContentContext context, AliasPart
                 templateContext.SetValue("ContentItem", part.ContentItem);
 
                 part.Alias = await _liquidTemplateManager.RenderAsync(pattern, NullEncoder.Default, templateContext);
-                part.Apply();
-            }
-        }
+                part.Alias = part.Alias.Replace("\r", String.Empty).Replace("\n", String.Empty);
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]:
```csharp
part.Alias = part.Alias.Replace ("\r\n", String.Empty);
```

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs [hunk 2]
```diff
@@ -90,9 +81,20 @@ public override async Task CloningAsync(CloneContentContext context, AliasPart p
             clonedPart.Alias = await GenerateUniqueAliasAsync(clonedPart.Alias, clonedPart);
 
             clonedPart.Apply();
+        }
+
+        /// <summary>
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: Comments for private method is not required

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs [hunk 3]
```diff
@@ -108,11 +110,16 @@ private async Task<string> GenerateUniqueAliasAsync(string alias, AliasPart cont
...
                 var versionedAlias = $"{unversionedAlias}-{version++}";
-                if ((await _session.QueryIndex<AliasPartIndex>(o => o.Alias == versionedAlias && o.ContentItemId != context.ContentItem.ContentItemId).CountAsync()) == 0)
+                if (await IsAliasUniqueAsync(versionedAlias, context))
                 {
                     return versionedAlias;
                 }
             }
         }
+
+        private async Task<bool> IsAliasUniqueAsync(string alias, AliasPart context)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: We can use inline lambda expression instead

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 1 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: Nope. won't work on linux if we do that.

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 2 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: straight out of autoroute. don't see the problem with it myself

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 3 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: check the code, the method is used twice

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: What about

```csharp
part.Alias = part.Alias.Replace (Environment.NewLine, String.Empty);
```

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 3 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: Even though

```csharp
private async Task<bool> IsAliasUniqueAsync(string alias, AliasPart context)
    => (await _session.QueryIndex<AliasPartIndex>(o => o.Alias == alias && o.ContentItemId != context.ContentItem.ContentItemId).CountAsync()) == 0;
```

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: https://docs.microsoft.com/en-us/dotnet/csharp/local-functions-vs-lambdas

I'm happy with it as a method.

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 1 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: Somehow I feel there may still be a problem introduced by using `Environment.NewLine` and adds little advantage

Also it is consistent with elsewhere in the code base (see the `AutorouteHandler`)

File: src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs (hunk 3 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: but it's not a local method :smile:

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs]: It's an expression bodied function that results to the same IL code ;)

It allows more compact code, but not sure it is always more readable and sometimes i don't use it when there is a good chance that i have to add a line of code and then to re-write the full body.

But yes in some contexts i like to use them. If it becomes for us a mandatory syntax rule, maybe could be part of a refactoring work, they are many places that could be updated.

But here i don't think it should be a blocking point.

**Pull request merged**